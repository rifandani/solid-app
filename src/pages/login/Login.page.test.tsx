import { Route } from '@solidjs/router';
import { waitFor } from '@solidjs/testing-library';
import { vi } from 'vitest';
import { renderWithRouter } from '../../utils/test.util';
import LoginPage from './Login.page';

vi.spyOn(localStorage, 'getItem');
localStorage.getItem = vi.fn(() => 'user');

describe('LoginPage', () => {
  it('should render correctly', async () => {
    // ARRANGE
    const { container, getByText } = renderWithRouter(() => (
      <Route path="/" component={LoginPage} /> // the path doesn't matter here
    ));

    // ASSERT
    await waitFor(() => {
      expect(getByText(/Welcome Back/)).toBeInTheDocument();
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
