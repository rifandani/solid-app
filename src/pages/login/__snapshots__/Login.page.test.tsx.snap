// Vitest Snapshot v1

exports[`LoginPage > should render correctly 1`] = `
<main
  class="h-screen bg-white"
>
  <div
    class="flex w-full flex-wrap"
  >
    <article
      class="flex w-full flex-col md:w-1/2"
    >
      <section
        class="flex justify-center pt-12 md:-mb-24 md:justify-start md:pl-12"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
          link=""
        >
          <a
            class="relative h-16 w-16 cursor-pointer"
          >
            <svg
              class="hover:text-violet-800 rounded-full"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 2a1 1 0 00-.71.297l-10.087 8.8A.5.5 0 001 11.5a.5.5 0 00.5.5H4v8a1 1 0 001 1h4a1 1 0 001-1v-6h4v6a1 1 0 001 1h4a1 1 0 001-1v-8h2.5a.5.5 0 00.5-.5.5.5 0 00-.203-.402l-10.08-8.795a1 1 0 00-.006-.006A1 1 0 0012 2z"
                fill="currentColor"
              />
            </svg>
          </a>
        </a>
      </section>
      <section
        class="my-auto flex flex-col justify-center px-8 pt-8 md:justify-start md:px-24 md:pt-0 lg:px-32"
      >
        <p
          class="text-center text-3xl"
        >
          Welcome Back
        </p>
        <form
          class="flex flex-col pt-3 md:pt-8"
        >
          <div
            class="flex flex-col pt-4"
          >
            <label
              class="text-lg"
              for="email"
            >
              Email
            </label>
            <input
              autofocus=""
              class="mt-1"
              name="email"
              placeholder="Your email..."
              required=""
              type="email"
            />
          </div>
          <div
            class="flex flex-col pt-4"
          >
            <label
              class="text-lg"
              for="password"
            >
              Password
            </label>
            <input
              class="mt-1"
              minlength="8"
              name="password"
              placeholder="Your password..."
              required=""
              type="password"
            />
          </div>
          
          <button
            class="font-semibold items-center justify-center inline-flex space-x-2 select-none cursor-pointer duration-200 text-white bg-violet-800 border-0 ring-violet-800 hover:bg-violet-500 active:bg-violet-200 focus:ring-4 disabled:opacity-50 rounded text-md py-4 px-4 mt-8"
            type="submit"
          >
            Login
          </button>
        </form>
        <div
          class="pt-12 pb-12 text-center"
        >
          <p>
            Don't have an account? 
            <a
              class="inactive"
              href="/register"
              link=""
            >
              <a
                class="cursor-pointer font-semibold hover:text-violet-800"
              >
                Register here.
              </a>
            </a>
          </p>
        </div>
      </section>
    </article>
    <article
      class="w-1/2 shadow-2xl"
    >
      <span
        class="relative hidden h-screen w-full md:flex md:justify-center md:items-center"
      >
        <img
          alt="login page cover"
          src="src/assets/online-payment.png"
        />
      </span>
    </article>
  </div>
</main>
`;

exports[`LoginPage > should render correctly 2`] = `
<main
  class="h-screen bg-white"
>
  <div
    class="flex w-full flex-wrap"
  >
    <section
      class="flex w-full flex-col md:w-1/2"
    >
      <div
        class="flex justify-center pt-12 md:-mb-24 md:justify-start md:pl-12"
      >
        <a
          aria-current="page"
          class="relative cursor-pointer active"
          href="/"
          link=""
        >
          <svg
            class="h-8 w-8 rounded-full hover:text-primary"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2a1 1 0 00-.71.297l-10.087 8.8A.5.5 0 001 11.5a.5.5 0 00.5.5H4v8a1 1 0 001 1h4a1 1 0 001-1v-6h4v6a1 1 0 001 1h4a1 1 0 001-1v-8h2.5a.5.5 0 00.5-.5.5.5 0 00-.203-.402l-10.08-8.795a1 1 0 00-.006-.006A1 1 0 0012 2z"
              fill="currentColor"
            />
          </svg>
        </a>
      </div>
      <div
        class="my-auto flex flex-col justify-center px-8 pt-8 md:justify-start md:px-24 md:pt-0 lg:px-32"
      >
        <p
          class="text-center text-3xl text-primary"
        >
          Welcome Back
        </p>
        <form
          class="form-control pt-3 md:pt-8"
        >
          <div
            class="form-control pt-4"
          >
            <label
              class="label"
              for="email"
            >
              <span
                class="label-text"
              >
                Email
              </span>
            </label>
            <input
              autofocus=""
              class="input-bordered input-primary input mt-1 shadow-md"
              name="email"
              placeholder="Your email..."
              required=""
              type="email"
            />
          </div>
          <div
            class="form-control pt-4"
          >
            <label
              class="label"
              for="password"
            >
              <span
                class="label-text"
              >
                Password
              </span>
            </label>
            <input
              class="input-bordered input-primary input mt-1 shadow-md"
              minlength="8"
              name="password"
              placeholder="Your password..."
              required=""
              type="password"
            />
          </div>
          
          <button
            class="btn-primary btn mt-8 normal-case"
            type="submit"
          >
            Login üéÅ
          </button>
        </form>
        <div
          class="py-12 text-center"
        >
          <p>
            Don't have an account? 
            <a
              class="link-primary link inactive"
              href="/register"
              link=""
            >
              Register here
            </a>
          </p>
        </div>
      </div>
    </section>
    <section
      class="w-1/2 shadow-2xl"
    >
      <span
        class="relative hidden h-screen w-full md:flex md:items-center md:justify-center"
      >
        <img
          alt="login page cover"
          src="src/assets/online-payment.png"
        />
      </span>
    </section>
  </div>
</main>
`;
